src/main/java/com/example/todolist/API/Restful/RestController/TaskController.java

```diff
@@ -86,4 +85,22 @@ public ResponseEntity<?> getTaskByTitle(@RequestParam(defaultValue = "", name="t
         Page<TaskResponse> pageTaskResponses = new PageImpl<>(taskResponses, paging, tasks.getTotalElements());
         return ResponseEntity.status(200).body(new ResponseWithData<>("",pageTaskResponses));
     }
+
+    @PostMapping("/complete")
+    public ResponseEntity<?> completeTask(@RequestParam("id") String id) {
+        var task = taskRepository.findByIdWithUser(id);
+
+        if (task == null) {
+            return ResponseEntity.status(404).body(new Response("Task not found!"));
+        }
+
+        task.setStatus(Status.DONE);
+        taskRepository.save(task);
+        TaskResponse taskResponse = TaskResponse.toDtoResponse(task);
+
+        String emailBody = "Your task \"" + task.getTitle() + "\" has been completed!";
+        emailSenderService.sendEmailForTaskCompleting(task.user.getEmail(), emailBody);
```

Comment:
since you name it sendEmailForTaskCompleting, you should not to pass the email body as a paramater, let's declare it where it should be